Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -0,0 +1,413 @@
+                                                                           // to just have a different 2D confiner
+                                                                           // for each setup
+        [HideInInspector, SerializeField] internal bool m_TriggerBake = false;
+        [HideInInspector, SerializeField] internal bool m_TriggerClearCache = false;
+        [HideInInspector, SerializeField] internal float m_MaxOrthoSize;
+        [HideInInspector, SerializeField] internal bool m_ShrinkToPointsExperimental;
+        
+        private static readonly float m_bakedConfinerResolution = 0.005f;
+        
+        internal enum BakeProgressEnum { EMPTY, BAKING, BAKED, INVALID_CACHE } // TODO: remove states after
+                                                                               // fist pass cleanup!
+        [HideInInspector, SerializeField] internal BakeProgressEnum BakeProgress = BakeProgressEnum.INVALID_CACHE;
+
+        private List<List<Vector2>> m_originalPath;
+        private List<List<Vector2>> m_originalPathCache;
+        private int m_originalPathTotalPointCount;
+        
+        private float m_frustumHeightCache;
+        private List<List<Vector2>> m_currentPathCache;
+
+        private List<ConfinerOven.ConfinerState> m_confinerStates;
+        private ConfinerOven m_confinerBaker = null;
+
+        /// <summary>Force rebake manually. This function invalidates the cache and rebakes the confiner.</summary>
+        public void ForceBake()
+        {
+            InvalidatePathCache();
+            Bake();
+        }
+        
+        /// <summary>
+        /// Trigger rebake process manually.
+        /// The confiner rebakes iff an input parameters affecting the outcome of the baked result change.
+        /// </summary>
+        private void Bake()
+        {
+            m_TriggerBake = true;
+        }
+
+        protected override void PostPipelineStageCallback(CinemachineVirtualCameraBase vcam, 
+            CinemachineCore.Stage stage, ref CameraState state, float deltaTime)
+        {
+            if (stage == CinemachineCore.Stage.Body)
+            {
+                if (!ValidateConfinerStateCache(state.Lens.Aspect, out bool confinerStateChanged))
+                {
+                    return; // invalid path
+                }
+                
+                float frustumHeight = CalculateFrustumHeight(state, vcam);
+                ValidatePathCache(confinerStateChanged, frustumHeight);
+
+                var extra = GetExtraState<VcamExtraState>(vcam);
+                Vector3 displacement = ConfinePoint(state.CorrectedPosition);
+                if (VirtualCamera.PreviousStateIsValid && deltaTime >= 0)
+                { 
+                    float displacementAngle = Vector2.Angle(extra.m_previousDisplacement, displacement);
+                    if (m_CornerDampingIsOn || m_Damping > 0 && displacementAngle > m_CornerAngleTreshold)
```

Comment:
does && have higher precedence than || ? I never remember 
(so brackets make sense here for clarity)